### PR TITLE
[fix][doc] Point Python and C++ docs to new GitHub repos

### PR DESCRIPTION
**Pull request by michaeljmarshall**
### Motivation

With the recent PIP to move the Python and C++ clients to their own GitHub repo, we need to update the documentation.

### Documentation

- [x] `doc`

### CODE REVIEW

File: site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md [hunk 1]
```diff
@@ -654,7 +654,7 @@ Client client("pulsar+ssl://my-broker.com:6651", config);
 
 ```
 
-For complete examples, refer to [C++ client examples](https://github.com/apache/pulsar/tree/master/pulsar-client-cpp/examples).
+For complete examples, refer to [C++ client examples](https://github.com/apache/pulsar-client-cpp/examples).
```
Comment (lhotari) [inline site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md:657]: this url needs to be fixed. please check that subdirectories are properly referenced.

**Review by lhotari — CHANGES_REQUESTED** [review]

File: site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md (hunk 1 above)
Comment (lhotari) [reply site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md:657]: it seems that `main` is the default branch now. so this url is https://github.com/apache/pulsar-client-cpp/tree/main/examples

Comment (michaeljmarshall) [reply site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md:657]: Great catch. It should be https://github.com/apache/pulsar-client-cpp/tree/main/examples

Comment (michaeljmarshall) [reply site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md]: Fixed

**Review by lhotari — APPROVED** [review]

**Comment by michaeljmarshall** [comment]
@merlimat @lhotari - I added links to redirect any users that have bookmarked the client directories

**Review by lhotari — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Although I don't think we still maintain historical documents, it's a historical issue that we use external (unstable) links in a versioned documents :)

+1 to merge.

**Pull request merged**